internal/provider/waypoint/data_source_waypoint_add_on_definition.go

```diff
@@ -0,0 +1,225 @@
+	return []datasource.ConfigValidator{
+		datasourcevalidator.Conflicting(
+			path.MatchRoot("name"),
+			path.MatchRoot("id"),
+		),
+	}
+}
+
+type DataSourceAddOnDefinition struct {
+	client *clients.Client
+}
+
+type DataSourceAddOnDefinitionModel struct {
+	ID                     types.String `tfsdk:"id"`
+	Name                   types.String `tfsdk:"name"`
+	ProjectID              types.String `tfsdk:"project_id"`
+	OrgID                  types.String `tfsdk:"organization_id"`
+	Summary                types.String `tfsdk:"summary"`
+	Labels                 types.List   `tfsdk:"labels"`
+	Description            types.String `tfsdk:"description"`
+	ReadmeMarkdownTemplate types.String `tfsdk:"readme_markdown_template"`
+
+	TerraformCloudWorkspace *tfcWorkspace    `tfsdk:"terraform_cloud_workspace_details"`
+	TerraformNoCodeModule   *tfcNoCodeModule `tfsdk:"terraform_no_code_module"`
+}
+
+func NewAddOnDefinitionDataSource() datasource.DataSource {
+	return &DataSourceAddOnDefinition{}
+}
+
+func (d *DataSourceAddOnDefinition) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_waypoint_add_on_definition"
+}
+
+func (d *DataSourceAddOnDefinition) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The Waypoint Add-on Definition data source retrieves information on a given Add-on Definition.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Optional:    true,
+				Description: "The ID of the Add-on Definition.",
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Add-on Definition.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "The ID of the HCP organization where the Waypoint Add-on Definition is located.",
+				Computed:    true,
+			},
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where the Waypoint Add-on Definition is located.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"summary": schema.StringAttribute{
+				Description: "A short summary of the Add-on Definition.",
+				Computed:    true,
+			},
+			"description": schema.StringAttribute{
+				Description: "A longer description of the Add-on Definition.",
+				Computed:    true,
+			},
+			"readme_markdown_template": schema.StringAttribute{
+				Computed:    true,
+				Description: "Instructions for using the definition (markdown format supported)",
+			},
+			"labels": schema.ListAttribute{
+				Computed:    true,
+				Description: "List of labels attached to this Add-on Definition.",
+				ElementType: types.StringType,
+			},
+			"terraform_cloud_workspace_details": &schema.SingleNestedAttribute{
```

Comment:
This is one thing that I agree with but it is already an established pattern so if we do want to make this change it will effect application templates as well.

Reply:
For what it's worth I'm fine with switching to `tfc` if we want. I'm also fine leaving it as `terraform`. I'm not sure I see much saving of typing time because these are fields in a configuration file as opposed to being repeatedly typed on the command line, but I wouldn't oppose ideas/attempts to improve user experience.

Reply:
Clint and I have decided that if we do make this change, it will be done in it's own PR because it effects additional files.